dclab/rtdc_dataset/writer.py

```diff
@@ -210,14 +216,33 @@ def store_feature(self, feat, data, shape=None):
                         + 'in the `info["feature shapes"]` key of '
                         + "your plugin feature.")
                     shape = data.shape[1:]
-            if shape == data.shape:
-                data = data.reshape(1, *shape)
-            elif shape == data.shape[1:]:
-                pass
+            # Condition for single or multiple events to be saved
+            if shape == data.shape or shape == data.shape[1:]:
```

Comment:
It would be much cleaner if you kept the original separation between single and multiple events. Combining them in one case complicates things a lot.

Reply:
How about introducing an `is_ragged` variable? Then, after these two original cases you decide based on the `shape` whether to write as ragged, image_grayscale, or ndarray (because you may also have 3d features)?

Reply:
> It would be much cleaner if you kept the original separation between single and multiple events. Combining them in one case complicates things a lot.

In that case, we would end up with a lot of conditions, right?